### PR TITLE
Upgrade Compose to 1.0.0-alpha09.

### PR DESCRIPTION
**Pull request by zach-klippenstein**

### CODE REVIEW

File: buildSrc/src/main/java/Dependencies.kt [hunk 1]
```diff
@@ -3,9 +3,9 @@ object Versions {
    * To change this in the IDE, use `systemProp.square.kotlinVersion=x.y.z` in your
    * `~/.gradle/gradle.properties` file.
    */
-  val KotlinCompiler = System.getProperty("square.kotlinVersion") ?: "1.4.10"
+  val KotlinCompiler = System.getProperty("square.kotlinVersion") ?: "1.4.21"
```
Comment (zach-klippenstein) [inline buildSrc/src/main/java/Dependencies.kt:6]: Latest compose version requires this kotlin version.

File: compose-tests/src/androidTest/java/radiography/test/compose/ComposeUiTest.kt [hunk 2]
```diff
@@ -46,15 +45,15 @@ import radiography.ViewStateRenderers.textViewRenderer
...
-@OptIn(ExperimentalRadiographyComposeApi::class, ExperimentalSubcomposeLayoutApi::class)
+@OptIn(ExperimentalRadiographyComposeApi::class)
 class ComposeUiTest {
 
   @get:Rule
   val composeRule = createComposeRule()
 
   @Test fun when_includingPii_then_hierarchyContainsText() {
     composeRule.setContentWithExplicitRoot {
-      Text("FooBar")
```
Comment (zach-klippenstein) [inline compose-tests/src/androidTest/java/radiography/test/compose/ComposeUiTest.kt:57]: `Text` is now material-specific.

File: compose-tests/src/androidTest/java/radiography/test/compose/ComposeUiTest.kt [hunk 3]
```diff
@@ -498,10 +497,12 @@ class ComposeUiTest {
...
-        LazyColumnFor(items = listOf(1, 2, 3), modifier = Modifier.testTag("list")) {
-          Box(Modifier.testTag("child:$it"))
-          if (it % 2 == 0) {
-            Box(Modifier.testTag("child:$it (even)"))
+        LazyColumn(Modifier.testTag("list")) {
+          items(listOf(1, 2, 3)) {
+            Box(Modifier.testTag("child:$it"))
+            if (it % 2 == 0) {
+              Box(Modifier.testTag("child:$it (even)"))
+            }
```
Comment (zach-klippenstein) [inline compose-tests/src/androidTest/java/radiography/test/compose/ComposeUiTest.kt:505]: Our code style changed, probably want to do an indentation pass over the whole codebase.

File: compose-tests/src/androidTest/java/radiography/test/compose/ComposeUiTest.kt [hunk 4]
```diff
@@ -573,9 +574,9 @@ class ComposeUiTest {
     modifier: Modifier,
     children: @Composable () -> Unit
   ) {
-    SubcomposeLayout<Unit>(modifier) { constraints ->
+    SubcomposeLayout(modifier) { constraints ->
```
Comment (zach-klippenstein) [inline compose-tests/src/androidTest/java/radiography/test/compose/ComposeUiTest.kt:577]: This isn't being suddenly inferred, the type parameter was actually removed.

File: radiography/src/main/java/radiography/internal/ComposeLayoutInfo.kt [hunk 5]
```diff
@@ -50,18 +51,19 @@ private fun Group.computeLayoutInfos(
...
-        )
-      }
+    .flatMap { it.tryGetComposers().asSequence() }
+    .map { subcomposer ->
+      ComposeLayoutInfo(
+        isSubcomposition = true,
+        name = name,
+        bounds = box,
+        modifiers = emptyList(),
+        children = subcomposer.compositionData.asTree().layoutInfos,
```
Comment (zach-klippenstein) [inline radiography/src/main/java/radiography/internal/ComposeLayoutInfo.kt:63]: `slotTable` is now private, replaced by these more limited-surface-area APIs. They're still not intended for public consumption, hence the need for the `@OptIn` above.

File: compose-tests/build.gradle.kts [hunk 6]
```diff
@@ -30,6 +30,11 @@ android {
     kotlinCompilerVersion = Versions.KotlinCompiler
     kotlinCompilerExtensionVersion = Versions.Compose
   }
+
+  packagingOptions {
```
Comment (rjrjr) [inline compose-tests/build.gradle.kts:34]: Could you add a comment about why this is needed?

File: radiography/src/main/java/radiography/internal/ComposeLayoutInfo.kt (hunk 5 above)
Comment (rjrjr) [reply radiography/src/main/java/radiography/internal/ComposeLayoutInfo.kt:63]: Worth saying so in a code comment, that `compositionData` specifically is "cheating."

**Review by rjrjr — APPROVED** [review]

File: compose-tests/build.gradle.kts (hunk 6 above)
Comment (zach-klippenstein) [reply compose-tests/build.gradle.kts:34]: It's the same reason it's always needed for most kotlin libraries, I can link to more info but it's pretty standard.

**Comment by zach-klippenstein** [comment]
The build is failing because the GitHub workflow overrides the kotlin version to an old one: https://github.com/square/radiography/blob/2e0dff64cf4b9a4ea3bd559f9f800ba18628902b/.github/workflows/android.yml#L15

@pyricau do you remember why that's there? I think we should remove it.

File: compose-tests/src/androidTest/java/radiography/test/compose/ComposeUiTest.kt (hunk 3 above)
Comment (zach-klippenstein) [reply compose-tests/src/androidTest/java/radiography/test/compose/ComposeUiTest.kt:505]: #118

**Pull request merged**